packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -119,10 +140,93 @@ export class SerializedStateManager {
 		} else {
 			const { baseSnapshot, snapshotBlobs } = this.pendingLocalState;
 			this.snapshot = { baseSnapshot, snapshotBlobs };
+			this.refreshSnapshot(supportGetSnapshotApi).catch((error) => {
```

Comment:
what prevents this from being called multiple times, leading to multiple promises running?

Reply:
Nothing prevents it but we only load once and the method is only called here so there shouldn't be any multiple promises. We could add a reentrancy flag in preparation for refreshing periodically but not really needed right now imo

Reply:
we should be defensive here such that if someone changes the usage doesn't break subtly, otherwise it could be easy to unknowingly introduce bugs.